### PR TITLE
Feature/Rohde&Schwarz HMP series

### PR DESCRIPTION
**Pull request by MarceloSemO**
Driver for the R&S HMP series (similar to driver for HMC series, but with some subtle differences)

### CODE REVIEW

File: qcodes_contrib_drivers/drivers/RohdeSchwarz/private/HMP.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        if self.parent.model_no > 4000:
+            return 10
+        elif self.parent.model_no == 2020 and self.channel == 1:
+            return 10
+        return 5
+
+    def send_cmd(self, param, value):
+        self.write(f"INSTrument:NSELect {self.channel:d}")
+        if value is None:
+            return self.ask(f"{self.get_scpi_command(param)}?")
```
Comment (jenshnielsen) [inline qcodes_contrib_drivers/drivers/RohdeSchwarz/private/HMP.py:65]: I would normally not make get_scpi_command a method. I would just make the dictionary a private member variable on the class and do something like `self.ask(f"{self._commands[param]}?")`
Alternatively you could pass the command in via another argument to send_cmd. That would have the advantage of including the command with the parameter definition but at the cost of slight repetition.

File: qcodes_contrib_drivers/drivers/RohdeSchwarz/private/HMP.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        commands = {
+            "set_voltage": "SOURce:VOLTage:LEVel:IMMediate:AMPLitude",
+            "set_current": "SOURce:CURRent:LEVel:IMMediate:AMPLitude",
+            "state": "OUTPut:STATe",
+            "voltage": "MEASure:SCALar:VOLTage:DC",
+            "current": "MEASure:SCALar:CURRent:DC",
+        }
+        return commands.get(param)
+
+    def get_power(self):
```
Comment (jenshnielsen) [inline qcodes_contrib_drivers/drivers/RohdeSchwarz/private/HMP.py:81]: I don't think the user should ever need to call this directly so I would make it protected with a name like `_get_power`

**Review by jenshnielsen — APPROVED** [review]
Thanks, I left a few minor suggestions for potential improvements

**Comment by MarceloSemO** [comment]
@jenshnielsen, your suggestions are implemented now and I have rebased the branch to incorporate the recent merges into the master branch. I think this feature is now also ready for merging into the master branch :)

**Comment by jenshnielsen** [comment]
Thanks @MarceloSemO I will merge it now

**Pull request merged**